### PR TITLE
[WIP] update additional bower dependencies in web console

### PR DESCRIPTION
**Pull request by benjaminapetersen**
NOTE: currently branched off of [my other PR for npm & some bower](https://github.com/openshift/origin/pull/8670).  Will rebase on master once that PR is merged.

**previously updated**
- patternfly/bootstrap

**already latest**
- kubernetes-topology-graph
- kubernetes-container-terminal
- kubernetes-label-selector
- openshift-object-describer
- json3 - latest

**updated**
- bumps angular & angular modules upto 1.3.20
- bump uril.js to 1.17.1
- bumps layout.attrs to latest w/@sg00dwin's flexbox bugs fixes

**wont update**
- selectize.js - next 0.12.0 includes breaking changes
- sifter.js - only included as a selectize dependency, to pin to a specific version
- microplugin.js - only included as a selectize dependency, to pin to a specific version
- lodash.js - 3.10.1 is latest before going to 4.x.x (breaking changes)
- moment - close to latest, eonasdan-bootstrap-datetimepicker not compatible with newer version of moment, little wary of updating this repo 
- es5-shim - some breaking changes if updated, changelog is unclear 

By 'wont update' I mean for this PR, I'd suggest we consider doing them as individual PRs in future if we want to update them due to potential for breaking changes.

@jwforres @spadgett keeping rest of bower separate from the NPM changes

### CODE REVIEW

**Comment by benjaminapetersen** [comment]
@gashcrumb attempted to update `hawtio-core` form `2.0.11` to the latest `2.0.36`, to pick up the [fix for this bug](https://bugzilla.redhat.com/show_bug.cgi?id=1285658), but it breaks the menu. Dropped back to `2.0.32` where the fix came in, but no luck.  Curious if you may be able to help me diagnose. Thx!

**Comment by benjaminapetersen** [comment]
Holding off on `fork_ami` for this one until the npm half gets through.

**Comment by benjaminapetersen** [comment]
The [pathseg issue](https://github.com/c3js/c3/issues/1566) was merged 4 days ago (May 1).  Looking at this a bit more before dropping `pathseg`.

File: assets/bower.json [hunk 1]
```diff
@@ -92,6 +97,14 @@
         "src-min-noconflict/theme-dreamweaver.js",
         "src-min-noconflict/theme-eclipse.js"
       ]
+    },
+    "uri.js" : {
+      "main" : [
```
Comment (jwforres) [inline assets/bower.json]: we had said we werent going to use main to effectively alter what we get from wiredep, that we were only going to use the exclude in gruntfile for that from now on

Comment (jwforres) [inline assets/bower.json]: the exception would be if the main was missing something that we needed to add

Comment (benjaminapetersen) [inline assets/bower.json]: Ah, hmm. thats right. will fix.

**Comment by benjaminapetersen** [comment]
@gashcrumb @jwforres fwiw, the URI.js bower file has not `uri.js` or `urijs` but `URIjs`, though the folder is `uri.js` and if you search it on bower.io, you get `urijs` (and an npm search gives both `URIjs` and `urijs`).  Heh.  

```javascript
{
  "name": "URIjs",
  "version": "1.14.1",
  "main": [
```
I went with `uri.js` for the bower.json in **openshift**, but the bower.json file in **hawtio-core-navigation** at the version we have been using has `uri.js`, however the latest hawtio-core-navigation has `urijs`.

woot.

**Comment by benjaminapetersen** [comment]
@sg00dwin feel free to give this a pull and see if you see any bower install errors.

**Comment by danmcp** [comment]
Need to move to the ui repo if still valid.

**Comment by benjaminapetersen** [comment]
Ah, thx. Already moved and merged.

**Pull request closed without merging**